tests/unit/test_charm.py

```diff
@@ -1683,45 +1668,53 @@ def test_push_tls_files_to_workload(self, _get_tls_files, _render_file, _update_
         _update_config.side_effect = [True, False, False, False]
 
         # Test when all TLS files are available.
-        self.assertTrue(self.charm.push_tls_files_to_workload())
-        self.assertEqual(_render_file.call_count, 3)
+        assert harness.charm.push_tls_files_to_workload()
+        assert _render_file.call_count == 3
 
         # Test when not all TLS files are available.
         for _ in range(3):
             _render_file.reset_mock()
-            self.assertFalse(self.charm.push_tls_files_to_workload())
-            self.assertEqual(_render_file.call_count, 2)
+            assert not (harness.charm.push_tls_files_to_workload())
+            assert _render_file.call_count == 2
```

Comment:
Why the number of calls changed from 3 to 2 in this case?

Reply:
I don't see a change in the number of calls when comparing to the version in the main branch. the assertion before the `for` loop is testing `call_count == 3` and the assertion inside the loop is testing `call_count == 2`.

Reply:
My bad. I looked at it wrongly. The change is right. Sorry for the confusion.